### PR TITLE
Use winrm_transport to set the correct default port for winrm

### PR DESCRIPTION
**Pull request by tim95030**
Fix for issue #178

### CODE REVIEW

File: lib/chef/knife/vsphere_vm_clone.rb [hunk 1]
```diff
@@ -368,15 +368,17 @@ def run
   def wait_for_access(connect_host, connect_port, protocol)
     if protocol == 'winrm'
       load_winrm_deps
+      if get_config(:winrm_transport) == 'ssl' && get_config(:winrm_port) == '5985'
+        config[:winrm_port] = '5986'
```
Comment (swalberg) [inline lib/chef/knife/vsphere_vm_clone.rb]: Thank for the patch. At first glance [this](https://github.com/chef/knife-windows/blob/master/lib/chef/knife/winrm_base.rb#L59) should be doing the work for you, is that not the case?

File: lib/chef/knife/vsphere_vm_clone.rb [hunk 2]
```diff
@@ -368,15 +368,17 @@ def run
...
+      end
       connect_port = get_config(:winrm_port)
-      print "\n#{ui.color('Waiting for winrm access to become available', :magenta)}"
+      print "\n#{ui.color("Waiting for winrm access to become available on #{connect_host}:#{connect_port}",:magenta)}"
```
Comment (swalberg) [inline lib/chef/knife/vsphere_vm_clone.rb]: Much better status message, thank you. :+1:

File: lib/chef/knife/vsphere_vm_clone.rb [hunk 3]
```diff
@@ -368,15 +368,17 @@ def run
...
       print('.') until tcp_test_winrm(connect_host, connect_port) do
         sleep 10
         puts('done')
       end
     else
-      print "\n#{ui.color('Waiting for sshd access to become available', :magenta)}"
-      # If FreeSSHd, winsshd etc are available
+      print "\n#{ui.color("Waiting for sshd access to become available on #{connect_host}:#{connect_port}", :magenta)}"
```
Comment (swalberg) [inline lib/chef/knife/vsphere_vm_clone.rb]: Pretty sure this addition would have saved me a bit of time over the past few months, thanks!

File: lib/chef/knife/vsphere_vm_clone.rb (hunk 1 above)
Comment (tim95030) [inline lib/chef/knife/vsphere_vm_clone.rb]: Except they added that to 1.0.0 and it didn't work with our code and I didn't have time to debug why. Not to mention they don't have an official release of 1.0.0 yet so this code is essentially a temporary workaround.

Comment (swalberg) [inline lib/chef/knife/vsphere_vm_clone.rb]: Ah, ok. We'll keep an eye upstream on this then.

**Pull request merged**